hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,225 @@
+- RCONT (corresponds to FIFO inputs {{< regref FDATA.RCONT >}}, only used with READ):
+If RCONT is set, the Format Byte represents part of a longer sequence of reads, allowing for reads to be chained indefinitely.
+The RCONT flag indicates the the final byte returned with the current read
+should be responded to which an ACK, allowing the target to continue sending data.
+(Note that the first R-1 bytes read will still be acknowledged regardless of whether RCONT is asserted or not.)
+- START (corresponds to {{< regref FDATA.START >}}, Ignored when used with READ):
+Issue a START condition before transmitting the Format Byte on the bus.
+This flag may also be used to issue a repeated start condition.
+- STOP (corresponds to {{< regref FDATA.STOP >}}):
+Issue a STOP signal after processing this current entry in the FMT FIFO.
+Note that this flag is not compatible with (READ & RCONT), and will cause bus conflicts.
+- NAKOK (corresponds to {{< regref FDATA.NAKOK >}}, Not compatible with READ):
+Typically every byte transmitted must also receive an ACK signal, and the IP will raise an exception if no ACK is received.
+However, there are some I2C commands which do not require an ACK.
+In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received. 
+
+### Timing Control Registers
+
+For standard mode, fast-mode and fast-mode plus, the timing requirements for each transaction are detailed in Table 10 of the [I2C specification](https://www.nxp.com/docs/en/user-guide/UM10204.pdf).
+In order to claim complete compatibility at each mode, the state machine timings need to be adapted to whether there are Standard-mode, Fast-mode and Fast-mode Plus devices on the bus.
+Furthermore, depending on the actual capacitance of the bus, even a bus with all Fast-mode Plus capable devices may have to operate at slower speeds than 1Mbaud.
+For example, the host may need to run at lower frequencies, as discussed in Section 5.2 of the specification, but the computation of the nominal frequency will depend on timing specifications in Table 10, in this case particularly, the limits on T<sub>LOW</sub>, T<sub>HIGH</sub>, T<sub>r</sub>, and T<sub>f</sub>.
+Assuming no clock stretching, for a given set of these four parameters the baud rate is then given to be:
+$$1/f_{SCL}=T_{LOW}+T_{HIGH}+T_{r}+T_{f}.$$ 
+
+Thus in order to ensure compliance with the spec in any particular configuration, software will program the I2C host IP with explicit values for each of the following timing parameters, as defined in Figure 38 of the specification.   
+- T<sub>LOW</sub>: set in register {{< regref TIMING0.TLOW >}}.
+- T<sub>HIGH</sub>: set in register {{< regref TIMING0.THIGH >}}.
+- T<sub>r</sub>: set in register {{< regref TIMING1.T_R >}}.
+(Note: The rise time cannot be explicitly controlled by internal hardware, and will be a function of the capacitance of the bus.
```

Comment:
this may just be me, but i tend to find an example or two very helpful when discussing setting of timing parameters.

Reply:
Hi @tjaychen, Alright at long last, these examples are in the latest commit (https://github.com/lowRISC/opentitan/pull/1047/commits/21fad6f116bb3589a6a65ddc40808f677a7ae6af)
Please take a look, and let me know if you have any comments or suggestions.